core/templates/dev/head/domain/topic/TopicObjectFactory.js

```diff
@@ -17,23 +17,28 @@
  * topic domain objects.
  */
 
-oppia.factory('TopicObjectFactory', ['SubtopicObjectFactory',
-  function(SubtopicObjectFactory) {
+oppia.factory('TopicObjectFactory', [
+  'SubtopicObjectFactory', 'SkillSummaryObjectFactory',
+  function(SubtopicObjectFactory, SkillSummaryObjectFactory) {
     var Topic = function(
         id, name, description, languageCode, canonicalStoryIds,
-        additionalStoryIds, uncategorizedSkillIds, nextSubtopicId, version,
-        subtopics) {
+        additionalStoryIds, uncategorizedSkillIds,
+        nextSubtopicId, version, subtopics, skillIdToDescriptionMap) {
```

Comment:
Seems like the subtopics here are subtopicBackendDicts?

Convert them into actual Subtopic objects before passing them into the Topic constructor.

Reply:
Done.